frontend/src/actionCreators/searchActionCreator.js

```diff
@@ -0,0 +1,62 @@
+import { showLoading, hideLoading } from "react-redux-loading-bar";
+import { request, success, error, clear } from "@/actions/genericActions";
+import * as reducerTypes from "@/constants/reducerTypes";
+import * as searchActions from "@/actions/searchActions";
+import * as API from "@/constants/API";
+import { ENVIRONMENT } from "@/constants/environment";
+import { createRequestHeader } from "@/utils/RequestHeaders";
+import CustomAxios from "@/customAxios";
+
+export const fetchSearchResults = (searchTerm, searchTypes) => (dispatch) => {
+  dispatch(request(reducerTypes.GET_SEARCH_RESULTS));
+  dispatch(showLoading("modal"));
+  return CustomAxios()
+    .get(
+      ENVIRONMENT.apiUrl + API.SEARCH({ search_term: searchTerm, search_types: searchTypes }),
+      createRequestHeader()
+    )
+    .then((response) => {
+      dispatch(success(reducerTypes.GET_SEARCH_RESULTS));
+      dispatch(searchActions.storeSearchResults(response.data));
+      dispatch(hideLoading("modal"));
+    })
+    .catch(() => dispatch(error(reducerTypes.GET_SEARCH_RESULTS)))
+    .finally(() => dispatch(hideLoading("modal")));
+};
+
+export const fetchSearchBarResults = (searchTerm) => (dispatch) => {
+  dispatch(request(reducerTypes.GET_SEARCH_BAR_RESULTS));
+  dispatch(showLoading("modal"));
```

Comment:
Does this loading bar make sense? (in any of these actionCreators) it is modal specific loading, and I don't think you are using models for the search.

Reply:
I didn't realize that's how it worked. I'll change that 👍

Reply:
Changed!